test/lib/ansible_test/_internal/sanity/yamllint.py

```diff
@@ -68,6 +70,16 @@ def test(self, args, targets, python_version):
         :type python_version: str
         :rtype: TestResult
         """
+        ansible_util.check_pyyaml(args, python_version)
+        if not ansible_util.CHECK_YAML_VERSIONS[python_version]['cloader']:
+            return SanityFailure(self.name, messages=[SanityMessage(
+                code='missing-compiled-pyyaml',
+                message='The yamllint sanity test needs PyYaml compiled with the libyaml backend to function',
+                path='',
+                line=0,
+                column=0,
+                level='error')])
```

Comment:
I realized last night my suggestion to make this an error is inconsistent with how the other sanity tests handle missing requirements. To maintain consistency it should be a warning instead:

```python
            display.warning("Skipping sanity test '%s' due to missing libyaml support in PyYAML." % self.name)
            return SanitySkipped(self.name)
```

I still see value in making missing test requirements an error, at least under some circumstances. We'll need to think more about how that should behave, and make sure it's consistent for all the tests.

Reply:
Done